src/aof.c

```diff
@@ -572,10 +573,29 @@ int writeAofManifestFile(sds buf) {
             tmp_am_name, am_name, strerror(errno));
 
         ret = C_ERR;
+        goto cleanup;
+    }
+
+    /* Also sync the AOF directory as new AOF files may be added in the directory */
+    dir_fd = open(server.aof_dirname, O_RDONLY);
```

Comment:
Do we do this before `rename` and `fsync` manifest ? If redis crash after `rename` and `fsync` manifest (before `fsync` directory), the newly opened AOF file may be missing from the directory (but recorded in manifest).

Reply:
After thinking about it, it seems that it is not necessary. Before `writeAofManifestFile`, we usually open a new INCR AOF. Assuming that the `dentry` corresponding to the manifest after rename has not been persisted, the `dentry` corresponding to the previously opened AOF has not been persisted either. There will be no problem if a crash occurs at this time.

The last `fsync` directory can ensure that the `dentry` corresponding to the newly opened INCR AOF and  renamed manifest are persisted.

Reply:
Yeah thank you, it's a issue I hadn't noticed before. In my opinion, there're indeed two things for directory to be sync, the AOF files referred by manifest, and the renamed manifest. As newly AOF file creation happens before manifest renaming and they're operating on the same directory,  the dirty directory flushing operated by OS in the background also follows the order. So the persistence for AOF file creation always doesn't happen after the manifest renaming, the intermediate status at any point would be integrated.